projects/hitl/software/hitl/iocontroller.py

```diff
@@ -1,302 +1,102 @@
-
-        self.log.info(f"Set state of {name} to {value}")
+        self.log.debug(f"Set state of {name} to {value}")
 
     def get_state(self, name: str) -> Union[int, float]:
         """Request a hardware state from the HitL system.
 
-        :param str name: The name of the state we want to get (e.x. "THROTTLE_POT_1", NOT 11)
-
-        :rtype:  Union[int, float]
-        :returns: The value of the requested state. If the signal is analog, returns a ``float``, otherwise an ``int``.
-
-        Getting an analog state is a 2 step process: we first write a command to the ADC to
-        start a conversion, then read the converted voltage back.
-
-        ADC command message format (1 byte):
-
-            * bits 0-2
-                * 101 (enable setting a new input channel, we always do this)
-            * bit 3:
-                * 1 (single-ended comparison, not differential)
-            * bits 4-7
-                * 0000 for channel 0
-                * 1000 for channel 1
-                * 0001 for channel 2
-                * 1001 for channel 3
-
-        ADC data returned message format (3 bytes):
-
-            * bit 0: 1 if voltage was positive (relative to ground)
-            * bit 1: 1 if reading was out of bounds
-            * bits 2-7: 0
-            * bits 8-23: 16 bit voltage reading
+        Args:
+            name (str): The name of the state to fetch (ex. "throttle_l_pos")
 
+        Returns:
+            (int/float): Value of the requested state - float for an analog, either
+                0 or 1 for digital
 
-        ADC datasheet: https://www.analog.com/media/en/technical-documentation/data-sheets/2489fb.pdf
         """
         # Raise an exception if the signal is write only
-        if self.pin[name]["read_write"] == "WRITE":
+        if self.pin[name]["read_write"] == PinMode.WRITE:
             raise Exception(f"{name} is a write-only signal, you cannot read it!")
 
-        # If no hardware, log an error
-        if not self.dev:
-            raise Exception("Could not get state, no hardware connection.")
-
-        # Create and send request
-        out = 0
         address = self.pin_info[name]["address"]
+        pin_number = self.pin_info[name]["pin"]
 
-        if self.pin_info[name]["type"] == "ANALOG":
-            # Request data
-            data = ADC_PREFIX & ADC_CHANNEL_TO_ADD_BITS[self.pin_info[name]["pin"]]
-            self.dev.i2cMaster_Write(address, data)
-
-            # Wait for response
-            response = self.dev.i2cMaster_Read(address, ADC_RETURN_SIZE_BYTES)
-            self.log.debug(f"Received {response}")
-            out = self._map_to_human(response, self.pin_info[name]["min"], self.pin_info[name]["max"])
+        if self.pin_info[name]["type"] == PinType.ANALOG:
+            min = self.pin_info[name]["min"]
+            max = self.pin_info[name]["max"]
+            out = self.ft4222.get_analog(address, pin_number, min, max)
         else:
```

Comment:
I would do `elif type == PinType.DIGITAL` and then have an `else` that raises a fault with invalid type.